skbio/tree/_nj.py

```diff
@@ -216,15 +220,25 @@ def _lowest_index(dm):
     method (#228).
 
     """
-    lowest_value = np.inf
-    for i in range(dm.shape[0]):
-        for j in range(i):
-            curr_index = i, j
-            curr_value = dm[curr_index]
-            if curr_value < lowest_value:
-                lowest_value = curr_value
-                result = curr_index
-    return result
+    # convert to an array
+    dmarray = np.array(dm.to_data_frame())
+    # convert to dense, 1-d squareform of distance matrix
+    vec = squareform(dmarray)
+    # get the positions of the lowest value
+    results = np.vstack(np.where(dmarray == np.amin(vec))).T
+    # select results in the bottom-left of the array
+    results = [r for r in results if r[0] > r[1]]
```

Comment:
perhaps inconsequential, but I believe this can also be:

```python
np.take(results, np.argwhere(results[:, 0] > results[:, 1]), axis=0).squeeze()
```